src/lasp_core.erl

```diff
@@ -269,6 +269,8 @@ declare(Id, Type, MetadataFun, MetadataNew, Store) ->
             %% Do nothing; make declare idempotent at each replica.
             {ok, {{Id, Type}, Type, Metadata, Value}};
         _ ->
+            %% @todo is this the correct place?
```

Comment:
Seems like `declare_if_not_found` calls  `declare/4` in core, so this change should be enough, right?